tools/flang2/flang2exe/lldebug.cpp

```diff
@@ -447,18 +487,23 @@ lldbg_create_subprogram_mdnode(
     llmd_add_md(mdb, def_context);
   llmd_add_i32(mdb, line);
   llmd_add_md(mdb, type_mdnode);
-  llmd_add_i1(mdb, is_local);
-  llmd_add_i1(mdb, is_definition);
+  if (!ll_feature_debug_info_ver90(&db->module->ir)) {
+    llmd_add_i1(mdb, is_local);
+    llmd_add_i1(mdb, is_definition);
+  }
   llmd_add_i32(mdb, virtuality);
   llmd_add_i32(mdb, vindex);
-  llmd_add_null(mdb);
+  llmd_add_null(mdb); // containingType field
   llmd_add_i32(mdb, flags);
-  llmd_add_i1(mdb, is_optimized);
+  if (ll_feature_debug_info_ver90(&db->module->ir))
+    llmd_add_i32(mdb, spFlags);
+  if (!ll_feature_debug_info_ver90(&db->module->ir))
```

Comment:
Could be an "else" branch?

Reply:
An `else` would make this a tad clearer to read. (The `!` is less clear.)

Reply:
Ack